File: Tree Data Structure/Tree Traversal/Python/Inorder Traversal.py
class tree:
    def __init__(self):
        self.data = int()
        self.left = None
        self.right = None

def createTree():
    head = tree()
    x = int(input("Enter data : "))
    head = create( x )
    if head != None:
        print("Left Child of",x," ", end="")
        head.left = createTree()
        print("Right Child of",x," ", end="")
        head.right = createTree()
    return head

def create(x):
    if x == -1:
        return None
    head = tree()
    head.data = x
    head.left = None
    head.right = None
    return head

def inorderDisp(root):
    if root.left != None:
        inorderDisp(root.left)
    print(root.data, " ", end ="")
    if root.right != None:
        inorderDisp(root.right)

root = createTree()
inorderDisp(root)

File: Tree Data Structure/Tree Traversal/C++/Inorder Traversal.cpp
#include <iostream>

using namespace std;

struct tree
{
    int data;
    tree* left;
    tree* right;
}*root;

tree* create( int i );
tree* createTree();
void inorder_disp( tree* root );

int main()
{
    root = createTree();
    inorder_disp( root );
    return 0;
}

tree* create( int i )
{
    tree* x = new tree;

    (*x).data = i;

    if( (*x). data == -1 )
    {
        return NULL;
    }

    (*x).left = NULL;
    (*x).right = NULL;

    return x;
}

tree* createTree()
{
    tree* head = new tree;

    int x;

    cout << "Enter data : ";
    cin >> x;

    head = create( x );
    if( head != NULL )
    {
        cout << "Left Child of " << x << " ";
        (*head).left = createTree();

        cout << "Right Child of " << x << " ";
        (*head).right = createTree();
    }

    return head;
}

void inorder_disp(tree* root )
{
    if( (*root).left != NULL )
    {
        inorder_disp( (*root).left );
    }

    cout << (*root).data << " ";

    if( (*root).right != NULL )
    {
        inorder_disp( (*root).right );
    }
}
